src/convertToObject.js

```diff
@@ -3,16 +3,46 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const properties = sourceString
+    .split('\n')
+    .filter((line) => line.includes(':'));
+
+  const divideProp = (item) => {
+    let items = item.split(':');
+
+    const preparationProp = (part) => {
+      if (part.includes(';')) {
+        return part.slice(0, -1).trim();
+      }
+
+      return part.trim();
+    };
+
+    items = items.map(preparationProp);
+
+    return {
+      [items[0]]: items[1],
+    };
+  };
+
+  const resultAraay = properties.map(divideProp);
+
+  return resultAraay.reduce((pre, item) => (
```

Comment:
too complicated, where it's unnecessary 
start by splitting by `;`, then each element by `:`, then just trim each element
then loop through the resulting array and assign key and value to prepared in advance empty object

Reply:
> then loop through the resulting array and assign key and value to prepared in advance empty object